require.js

```diff
@@ -87,11 +83,20 @@
     if (typeof bundle === 'object') {
       for (var key in bundle) {
         if (has.call(bundle, key)) {
-          modules[key] = bundle[key];
+          require.register(modules[key], bundle[key]);
         }
       }
     } else {
       modules[bundle] = fn;
+      if (extExp.test(bundle)) {
+        var woExt = bundle.replace(extExp, '');
+        var existing = aliases[woExt];
+        if (!existing || existing !== woExt + '.js') aliases[woExt] = bundle;
+      }
+      if (indexExp.test(bundle)) {
+        var woIndex = bundle.replace(indexExp, '');
+        if (!has.call(aliases, woIndex)) aliases[woIndex] = bundle;
+      }
```

Comment:
This does not take into account that the original file could be coffee, typescript, or whatever really. These files will only get `p/a/t/h.coffee` and not `p/a/t/h`